x-pack/plugins/cases/public/components/all_cases/all_cases_list.tsx

```diff
@@ -112,7 +108,10 @@ export const AllCasesList = React.memo<AllCasesListProps>(
 
     const sorting = useMemo(
       () => ({
-        sort: { field: queryParams.sortField, direction: queryParams.sortOrder },
+        sort: {
+          field: queryParams.sortField ?? DEFAULT_QUERY_PARAMS.sortField,
```

Comment:
Should we have `useUrlState` always return a valid `sortField` and `direction`? Meaning it would handle the defaulting for us instead of doing it here?

Reply:
The hook already handles the default, this is only so Typescript does not complain.

The problem is that `queryParams` has the type `UrlQueryParams` where:
```
  sortField?: SortFieldCase;
  sortOrder?: 'asc' | 'desc';
```

In practice, these fields will always be defined. The problem is that we need to initialize this object as empty because we don't just want the "default values" ( as in `useState<UrlQueryParams>(DEFAULT_QUERY_PARAMS);`).

To calculate those default values we need that logic where we first check the URL params, then the local storage, and finally we fall back to the DEFAULT constant.

So queryParams is **possibly** undefined, just for a very short while inside the hook initialization 😄